test/jdk/javax/swing/plaf/basic/BasicTextUI/8001470/bug8001470.java

```diff
@@ -26,45 +26,76 @@
  * @key headful
  * @bug 8001470
  * @summary JTextField's size is computed incorrectly when it contains Indic or Thai characters
- * @author Semyon Sadetsky
  */
 
-import javax.swing.*;
-import java.awt.*;
+import java.awt.Font;
+import java.awt.GraphicsEnvironment;
+import java.awt.GridLayout;
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+import javax.swing.JPanel;
 
 public class bug8001470 {
 
     private static JFrame frame;
     private static JTextField textField1;
     private static JTextField textField2;
+    private static boolean fontFound = false;
+
+    private static volatile int height1;
+    private static volatile int height2;
 
     public static void main(String[] args) throws Exception {
-        SwingUtilities.invokeAndWait(new Runnable() {
-            @Override
-            public void run() {
-                frame = new JFrame("JTextField Test");
-                frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        try {
+            SwingUtilities.invokeAndWait(new Runnable() {
+                @Override
+                public void run() {
+                    GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+                    String[] names = ge.getAvailableFontFamilyNames();
+                    Font[] allFonts = ge.getAllFonts();
+                    for (int x = 0; x < allFonts.length; x++) {
+                        if ((allFonts[x].canDisplay('\u0e01')) && (allFonts[x].canDisplay('\u0c01'))) {
+                            fontFound = true;
+                        }
+                    }
+                    if (!fontFound) {
+                        return;
+                    }
+                    frame = new JFrame("JTextField Test");
+                    frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
 
-                JPanel container = (JPanel) frame.getContentPane();
-                container.setLayout(new GridLayout(2,1));
+                    JPanel container = (JPanel) frame.getContentPane();
+                    container.setLayout(new GridLayout(2,1));
 
-                textField1 = new JTextField("\u0e01");
-                textField2 = new JTextField("\u0c01");
+                    textField1 = new JTextField("\u0e01");
+                    textField2 = new JTextField("\u0c01");
```

Comment:
~~Wouldn't it be enough to call `getPreferredSize()`?~~

No; if you call `getPreferredSize()`, the bug [JDK-8001470](https://bugs.openjdk.org/browse/JDK-8001470) can't be reproduced.